tests/test_algorithm.py

```diff
@@ -2018,3 +2022,99 @@ def test_remove_data(self):
         # initially only data for X should be sent and on the last day only
         # data for Y should be sent since X is expired
         np.testing.assert_array_equal(self.algo.data, expected_lengths)
+
+
+class TestAssetAutoClose(TestCase):
+    """
+    Tests if delisted securities are properly removed from a portfolio holding
+    positions in said securities.
+    """
+    def setUp(self):
+        dt = pd.Timestamp('2015-01-05', tz='UTC')
+        self.env = TradingEnvironment()
+        ix = self.env.trading_days.get_loc(dt)
+
+        self.metadata = make_rotating_equity_info(
```

Comment:
I'm inclined to say we should move the metadata construction here into a new test_utils function called `make_jagged_equity_info` or somesuch.